driver-core/src/main/java/com/datastax/driver/core/Fetchable.java

```diff
@@ -0,0 +1,166 @@
+     * fetched at any given time. If this method is called twice and the query
+     * triggered by the first call has not returned yet when the second one is
+     * performed, then the 2nd call will simply return a future on the currently
+     * in progress query.
+     *
+     * @return a future on the completion of fetching the next page of results.
+     * If the result set is already fully retrieved ({@code isFullyFetched() == true}),
+     * then the returned future will return immediately but not particular error will be
+     * thrown (you should thus call {@link #isFullyFetched()} to know if calling this
+     * method can be of any use}).
+     */
+    ListenableFuture<S> fetchMoreResults();
+
+    /**
+     * Returns the next result from this result set.
+     *
+     * @return the next row in this result set or null if this result set is
+     * exhausted.
+     */
+    T one();
+
+    /**
+     * Returns all the remaining rows in this result set as a list.
+     * <p/>
+     * Note that, contrary to {@link #iterator()} or successive calls to
+     * {@link #one()}, this method forces fetching the full content of the result set
+     * at once, holding it all in memory in particular. It is thus recommended
+     * to prefer iterations through {@link #iterator()} when possible, especially
+     * if the result set can be big.
+     *
+     * @return a list containing the remaining results of this result set. The
+     * returned list is empty if and only the result set is exhausted. The result set
+     * will be exhausted after a call to this method.
+     */
+    List<T> all();
+
+    /**
+     * Returns an iterator over the rows contained in this result set.
+     * <p/>
+     * The {@link Iterator#next} method is equivalent to calling {@link #one}.
+     * So this iterator will consume results from this result set and after a
+     * full iteration, the result set will be empty.
+     * <p/>
+     * The returned iterator does not support the {@link Iterator#remove} method.
+     *
+     * @return an iterator that will consume and return the remaining rows of
+     * this result set.
+     */
+    Iterator<T> iterator();
+
+    /**
+     * Returns information on the execution of the last query made for this result set.
+     * <p/>
+     * Note that in most cases, a result set is fetched with only one query, but large
+     * result sets can be paged and thus be retrieved by multiple queries. In that
+     * case this method return the {@link ExecutionInfo} for the last query
+     * performed. To retrieve the information for all queries, use {@link #getAllExecutionInfo}.
+     * <p/>
+     * The returned object includes basic information such as the queried hosts,
+     * but also the Cassandra query trace if tracing was enabled for the query.
+     *
+     * @return the execution info for the last query made for this result set.
+     */
+    ExecutionInfo getExecutionInfo();
```

Comment:
I thought the same, but went for placing them here 1) for practical reasons (both `ResultSet` and `Result` expose them) and 2) thinking that if you are fetching results, you are implicitly sending requests, so it's not so awkward after all to expose these methods here. But I'm fine with removing if everyone agrees.